src/components/CheckEditor/checkFormTransformations.ts

```diff
@@ -0,0 +1,339 @@
+import { SelectableValue } from '@grafana/data';
+import {
+  CheckType,
+  DnsRecordType,
+  DnsProtocol,
+  CheckFormValues,
+  Settings,
+  SettingsFormValues,
+  PingSettingsFormValues,
+  IpVersion,
+  PingSettings,
+  HttpSettings,
+  HttpMethod,
+  HttpSettingsFormValues,
+  Label,
+  TcpSettingsFormValues,
+  TcpSettings,
+  DnsSettingsFormValues,
+  DnsSettings,
+  DNSRRValidator,
+  DnsValidationFormValue,
+  ResponseMatchType,
+  Check,
+} from 'types';
+
+import { CHECK_TYPE_OPTIONS, IP_OPTIONS, DNS_RESPONSE_CODES } from 'components/constants';
+import { checkType } from 'utils';
+
+export function selectableValueFrom<T>(value: T, label?: string): SelectableValue<T> {
+  const labelValue: unknown = value;
+  return { label: label ?? (labelValue as string), value };
```

Comment:
perhaps `String(labelValue)` is safer here

Reply:
🎉 I was fighting the types here